tokio-postgres/src/connect.rs

```diff
@@ -374,14 +378,20 @@ async fn check_and_refresh(config: &Config) -> bool {
     } else {
         let duration = refresh_time.elapsed();
         if duration > config.yb_servers_refresh_interval {
-            let host_to_port_map = HOST_TO_PORT_MAP.lock().unwrap().clone();
+            let host_to_port_map = {
+                let host_to_port_map = HOST_TO_PORT_MAP.lock().unwrap();
+                host_to_port_map.clone()
+            };
             let mut index = 0;
             while index < host_list.len() {
                 let host = host_list.get(index);
                 let mut conn_host = host.unwrap().to_owned();
                 //check if we are to use public hosts
                 if USE_PUBLIC_IP.load(Ordering::SeqCst) {
-                    let public_host_map = PUBLIC_HOST_MAP.lock().unwrap().clone();
+                    let public_host_map = {
+                        let public_host_map = PUBLIC_HOST_MAP.lock().unwrap();
+                        public_host_map.clone()
+                    };
```

Comment:
This guard wasn't held over an await point but we changed it to drop the guard after the clone anyways because the guard isn't needed after the data is cloned. Best to close guards as soon as possible.

Reply:
Same as above.

Reply:
Removed unnecessary scope to drop. Fixed.